src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
That's true, where this matters is inside samplers like `IS`, which will repetitively call `model` to generate samples and compute importance weights.